autogluon/task/text_classification/dataset.py

```diff
@@ -119,28 +122,167 @@ def get_dataset(self, segment='train'):
         sampler, _ = get_split_samplers(dataset, split_ratio=0.2)
         return SampledDataset(dataset, sampler)
 
-class TSVClassificationTask(AbstractGlueTask):
-    def __init__(self, *args, **kwargs): # passthrough arguments to TSVDataset
-        # (filename, field_separator=nlp.data.Splitter(','), num_discard_samples=1, field_indices=[2,1])
-        self.args = args
-        self.kwargs = kwargs
-        is_pair = False
-        class_labels = ['0', '1']
-        metric = CompositeEvalMetric()
-        metric.add(F1())
-        metric.add(Accuracy())
-        super(TSVClassificationTask, self).__init__(class_labels, metric, is_pair)
-        dataset = nlp.data.TSVDataset(*self.args, **self.kwargs)
-        # do the split
-        train_sampler, val_sampler = get_split_samplers(dataset, split_ratio=0.8)
-        self.trainset = SampledDataset(dataset, train_sampler)
-        self.valset = SampledDataset(dataset, val_sampler)
+class CustomTSVClassificationTask(AbstractGlueTask):
+    """
+
+    Parameters
+    ----------
+    filepath: str, path of the file
+        Any valid string path is acceptable.
+    sep : str
+        Delimiter to use.
```

Comment:
Shouldn't `sep = '\t'` by default, given that this is called "TSV" dataset?

Reply:
Probably ',' , I think csv is generally more common. I don't think this should necessarily be called "TSV" if it supports all kinds of delimited files. I honestly don't think the name of the task should even mention the method of file loading.

Reply:
I'm just saying if the name of it TSV, then "\t" should be the default (I'm happy for it also to be CSV with "," as the default).  I think TSV format is more common for text data because commas appear way more frequently than tabs in natural language.  But I agree the method of file-loading probably does not need to be in the name of the class

Reply:
Would detail the data format and corresponding the sep in the doc.